From 9f5263b1e0598e0934877dc54b1bfc6107af04b8 Mon Sep 17 00:00:00 2001
From: Alona Solntseva <alona.solntseva@tandemg.com>
Date: Sun, 10 Apr 2022 15:27:43 +0300
Subject: [PATCH] [SW-67516] tpm: change pcr bank to sha256

Change the destination pcr bank of the pcr_extend() function
to sha256 bank

Signed-off-by: Alona Solntseva <alona.solntseva@tandemg.com>
Change-Id: I15effefde0d991c061646644863abcfab75a2bcb
---

diff --git a/arch/gaudi2/common/tpm_auth.c b/arch/gaudi2/common/tpm_auth.c
index 061e52b..d3e801e 100644
--- a/arch/gaudi2/common/tpm_auth.c
+++ b/arch/gaudi2/common/tpm_auth.c
@@ -95,7 +95,7 @@
 	if (hash == NULL)
 		return -1;
 
-	ret = tpm2_pcr_extend(image_index, TPM2_ALG_SHA384, hash, TPM2_SHA384_DIGEST_SIZE);
+	ret = tpm2_pcr_extend(image_index, TPM2_ALG_SHA256, hash, TPM2_SHA384_DIGEST_SIZE);
 	if (ret) {
 		preboot_puts("preboot_pcr_extend: pcr extend failed\n");
 		cpu_boot_err_set(CPU_BOOT_ERR_TPM_FAIL);
